src/file_handler.rs

```diff
@@ -369,4 +351,44 @@ mod test {
         let read_value = file_handler.read_file().expect("failed reading file");
         assert_eq!(read_value, write_value1);
     }
+
+    #[test]
+    fn concurrent_writes() {
+        use std::iter;
+        use std::sync::{Arc, Barrier};
+        use std::thread;
+
+        const NUM_THREADS : usize = 100;
+        const DATA_SIZE : usize = 10000;
+        const FILE_NAME : &'static str = "test2.json";
+
+        let _cleaner = ScopedUserAppDirRemover;
+        let barrier = Arc::new(Barrier::new(NUM_THREADS));
+
+        let handles = (0..NUM_THREADS).map(|i| {
+            let barrier = barrier.clone();
```

Comment:
The barrier synchronizes the threads to start writing to the file at the same time. So it actually increases the likelihood that the file will be written to concurrently.